crates/storage/provider/src/traits/block.rs

```diff
@@ -195,3 +197,55 @@ pub trait BlockProviderIdExt: BlockProvider + BlockIdProvider + ReceiptProviderI
     /// Returns `None` if block is not found.
     fn ommers_by_id(&self, id: BlockId) -> Result<Option<Vec<Header>>>;
 }
+
+/// BlockExecution Writer
```

Comment:
Ideally this would be an extension trait of `BlockReader + BlockWriter` given it reads *or* writes (takes)